src/realm/array_fixed_bytes.hpp

```diff
@@ -16,17 +16,20 @@
  *
  **************************************************************************/
 
-#ifndef REALM_ARRAY_OBJECT_ID_HPP
-#define REALM_ARRAY_OBJECT_ID_HPP
+#ifndef REALM_ARRAY_FIXED_BYTES_HPP
+#define REALM_ARRAY_FIXED_BYTES_HPP
 
 #include <realm/array.hpp>
 #include <realm/object_id.hpp>
+#include <realm/uuid.hpp>
 
 namespace realm {
 
-class ArrayObjectId : public ArrayPayload, protected Array {
+template <class ObjectType, int ElementSize>
```

Comment:
Would ElementSize not always be sizeof(ObjectType)?

Reply:
I think that would be a mistake, because it ties the C++ object representation to the file format - if I am understanding this correctly. @ironage ?

Reply:
Yes, the intention here was to leave the accessor class free to change without affecting the underlying storage size requirements. While it is true that right now the size will be equal to sizeof(ObjectType) I wanted to express that this may change in the future.

Reply:
OK - got it